lang/clang-11-bootstrap/Portfile

```diff
@@ -0,0 +1,275 @@
+        patchfiles-append \
+                    5004-patch-libcxx-reexport-symbols-lionorgreater-only.diff
+
+        # the libcxx installed by MacPorts on darwin 10 and less does not correspond to Apple's builtin libcxx
+        patchfiles-append \
+                    5005-MacPorts-only-patch-libcxx-includes-disable-availability-tests.diff
+    }
+
+    if {${os.major} <= 9} {
+        # TO DO: the blocks functionality can be replaced by libblocksruntime
+        # this might be integrated into clang on < 10.6
+        patchfiles-append \
+                    leopard-no-blocks.patch
+    }
+}
+
+configure.args-append \
+                    -DLIBCXX_ENABLE_SHARED=OFF \
+                    -DLLVM_BINDINGS_LIST=none \
+                    -DLLVM_BUILD_TOOLS=OFF \
+                    -DLLVM_ENABLE_PROJECTS=clang \
+                    -DLLVM_INCLUDE_EXAMPLES=OFF \
+                    -DLLVM_INCLUDE_TESTS=OFF \
+                    -DLLVM_TARGETS_TO_BUILD="AArch64\;X86" \
+                    -DPYTHON_EXECUTABLE=${prefix}/libexec/python27-bootstrap/bin/python2.7
+
+if {${os.major} <= 18} {
+    # on systems that might build i386, we need atomic builtins
+    # https://trac.macports.org/ticket/58712
+    configure.args-append \
+                    -DCOMPILER_RT_EXCLUDE_ATOMIC_BUILTIN=OFF
+}
+
+if {${os.major} < 11} {
+    variant emulated_tls description { enable c11/c++11 thread_local_storage support on older systems using emulated-tls } {
+        # use emulated-tls to support thread_local on systems prior to 10.7
+        patchfiles-append \
+                    9000-patch-llvm-7.0-support-emulated-tls.diff \
+                    9000-patch-clang-7.0-support-emulated-tls.diff
+    }
+
+    default_variants-append \
+                    +emulated_tls
+}
+
+# Be default Catalina and Mojave support a single arch at MacPorts
+if {${os.major} < 18 || ${os.major} > 19} {
+    default_variants-append     +universal
+}
+
+
+pre-configure {
+    file mkdir ${workpath}/bins
+
+    set gcc [open "${workpath}/bins/gcc" w 0755]
+    puts ${gcc} "#!/bin/sh"
+    puts ${gcc} "arch -arch $\{BUILD_ARCH:-${build_arch}\} ${prefix}/libexec/gcc10-bootstrap/bin/gcc \$@"
+    close ${gcc}
+
+    set gxx [open "${workpath}/bins/g++" w 0755]
+    puts ${gxx} "#!/bin/sh"
+    puts ${gxx} "arch -arch $\{BUILD_ARCH:-${build_arch}\} ${prefix}/libexec/gcc10-bootstrap/bin/g++ \$@"
+    close ${gxx}
+}
+
+configure.cc        ${workpath}/bins/gcc
+configure.cxx       ${workpath}/bins/g++
+
+if {${universal_possible} && [variant_isset universal]} {
+    foreach arch ${universal_archs_supported} {
+        lappend merger_configure_env(${arch}) BUILD_ARCH=${arch}
+        lappend merger_build_env(${arch}) BUILD_ARCH=${arch}
+    }
+}
+
+merger_arch_flag            yes
+merger_arch_compiler        yes
+merger_must_run_binaries    yes
+
+post-destroot {
+    system "ditto ${worksrcpath}/projects/libcxx/include ${destroot}${cmake.install_prefix}/lib/c++/v1"
```

Comment:
I still don't understand _why_ this was even in the normal clang ports as they should be installed into /include not /lib, but those should be installed correctly by some install target anyway

Reply:
@Gcenx to be honest I've just grab it from original port :)

Reply:
The original port has the headers in both /include and /lib due to the used install target I'm assuming, unless someone can explain _why_ there's a second copy in /lib it doesn't seem to be needed after all the testing I've done with this already.

Reply:
@Gcenx without this line I can't compile anything by this complier:
```
catap@Kirills-mini-m1 ~ % port contents clang-11-bootstrap | grep 'iostream'
catap@Kirills-mini-m1 ~ % cd /tmp 
catap@Kirills-mini-m1 /tmp % cat > test.cc
#include <iostream>
using namespace std;
int main() {
   cout << "Hello, World!" << endl; // This prints Hello, World!
   return 0;
}
catap@Kirills-mini-m1 /tmp % /opt/local/libexec/clang-11-bootstrap/bin/clang++ test.cc 
test.cc:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
catap@Kirills-mini-m1 /tmp % 
```

anyway, I agree that path looks quite wrong, special that `clang` is looking into `/opt/local/libexec/clang-11-bootstrap/bin/../include/c++/v1`

Reply:
Make this section

```
post-destroot {
    system "ditto ${worksrcpath}/projects/libcxx/include ${destroot}${cmake.install_prefix}/lib/c++/v1"
    file copy ${prefix}/libexec/gcc10-bootstrap/lib/libstdc++.6.dylib ${destroot}${cmake.install_prefix}/lib
}
```

That _should_ make it so `gcc10-bootstrap` isn't a runtime dep anymore, at least it worked during my testing.